### PR TITLE
refactor: implement irange::range and use it

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

File: src/util/irange.h [hunk 1]
```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+// Original implemtation here: https://github.com/whoshuu/cpp_range under MIT software license
```
Comment (kittywhiskers) [inline src/util/irange.h:4]: nit: correct spelling of implementation

File: src/util/irange.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+template <typename T>
+detail::Range<T> range(const T start, const T stop, const T step) {
+    return detail::Range<T>(start, stop, step);
+}
+
+} // namespace irange
+
+#endif // BITCOIN_UTIL_IRANGE_H
```
Comment (kittywhiskers) [inline src/util/irange.h:90]: nit: missing trailing newline

File: src/util/irange.h [hunk 3]
```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+// Original implemtation here: https://github.com/whoshuu/cpp_range under MIT software license
+
+
```
Comment (kittywhiskers) [inline src/util/irange.h:6]: nit: extra spacing

File: src/util/irange.h [hunk 4]
```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021 The Dash Core developers
```
Comment (kittywhiskers) [inline src/util/irange.h:1]: along with referring original implementation as a source, adding their copyright is probably for the best, also mention the commit of cpp_range from which this was derived from

Comment (PastaPastaPasta) [reply src/util/irange.h:1]: They didn't have a copyright on the file, just a MIT license in the folder

**Comment by PastaPastaPasta** [comment]
see latest

File: src/llmq/commitment.cpp [hunk 5]
```diff
@@ -80,16 +81,16 @@ bool CFinalCommitment::Verify(const CBlockIndex* pQuorumBaseBlockIndex, bool che
...
     }
     std::stringstream ss2;
-    for (size_t i = 0; i < llmq_params.size; i++) {
+    for (const auto i : irange::range(llmq_params.size)) {
         ss2 << "s[" << i << "]=" << signers[i];
     }
 
     LogPrintfFinalCommitment("CFinalCommitment::%s mns[%d] validMembers[%s] signers[%s]\n", __func__, members.size(), ss.str(), ss2.str());
-    for (size_t i = members.size(); i < size_t(llmq_params.size); i++) {
+    for (const auto i : irange::range(llmq_params.size)) {
```
Comment (PastaPastaPasta) [inline src/llmq/commitment.cpp:93]:
```suggestion
    for (const auto i : irange::range(members.size(), llmq_params.size)) {
```

File: src/llmq/signing_shares.cpp [hunk 6]
```diff
@@ -16,6 +16,7 @@
 #include <net_processing.h>
 #include <netmessagemaker.h>
 #include <spork.h>
+#include> util / irange.h>
```
Comment (PastaPastaPasta) [inline src/llmq/signing_shares.cpp:19]:
```suggestion
#include <util/irange.h>
```

File: src/llmq/quorums.cpp [hunk 7]
```diff
@@ -725,7 +726,10 @@ void CQuorumManager::StartCachePopulatorThread(const CQuorumCPtr pQuorum) const
 
     // when then later some other thread tries to get keys, it will be much faster
     workerPool.push([pQuorum, t, this](int threadId) {
-        for (size_t i = 0; i < pQuorum->members.size() && !quorumThreadInterrupt; i++) {
+        for (const auto i : irange::range(pQuorum->members.size())) {
+            if (!quorumThreadInterrupt) {
```
Comment (UdjinM6) [inline src/llmq/quorums.cpp:730]:
```suggestion
            if (quorumThreadInterrupt) {
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by UdjinM6** [comment]
pls see 49f76c6650

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**